server.js

```diff
@@ -553,38 +524,41 @@ app.post('/sortlist', function(req, res) {
     );
 });
 
-/* User requests a question; look it up by ID and store it in session. */
-app.post('/questionreq', function(req, res) {
+/* Display the question page. */
+app.get('/question', function(req, res) {
+    /* if the user has not yet logged in, redirect to login page */
     if (!req.session.user) {
         return res.redirect('/');
     }
 
-    questions.lookupQuestionById(parseInt(req.body.id), function(err, result) {
-        if (err) {
+    questions.lookupQuestionById(parseInt(req.query.id), function(err, questionFound) {
+        if (err || !questionFound) {
             return res.status(500).send();
         }
 
-        req.session.question = result;
-        if (req.session.user.answered.indexOf(result.id) == -1) {
-            req.session.questionAnswered = false;
-        } else {
-            req.session.questionAnswered = true;
+        if (!questionFound.visible) {
+            return res.status(400).send('Question is not available');
         }
 
-        return res.status(200).send();
+        return res.status(200).render('question', {
+            user: req.session.user,
+            question: questionFound,
+            answered: !(questionFound.answered.indexOf(req.session.user.id) === -1),
+            preview: false
+        });
     });
 });
 
 /* check if the submitted answer is correct */
-app.post('/submitanswer', function(req, res) {
+app.get('/submitanswer', function(req, res) {
```

Comment:
Same thing here, keep it as a post if you wanna be more legit.

Reply:
my bad